### PR TITLE
Make collection of non-existent files more clear

### PR DESCRIPTION
**Pull request by pyrco**
(DIS-962)

### CODE REVIEW

File: acquire/collector.py [hunk 1]
```diff
@@ -341,6 +341,9 @@ def collect_path(
             path = self.target.fs.path(path)
 
         try:
+            # If a path does not exist, is_dir(), is_file() and is_symlink() will return False (and not raise an
+            # exception), so we need to explicitly check for this.
+            exists = path.exists()
```
[the remark this answers is not in the document]
Comment (pyrco) [reply acquire/collector.py]: I keep `self.target.fs.lexists(path)` as there are some issues with `path.get().lexists()` and the underlying `FileEntry.get()` being implemented inconsistently. Some implementations do resolve and others do not resolve symlinks.

**Comment by pyrco** [comment]
@Schamper  Note: I added calls to reporting when a symlink or an unknown file type are collected (they are now reported as a failed collection).

File: acquire/collector.py [hunk 2]
```diff
@@ -341,6 +341,9 @@ def collect_path(
             path = self.target.fs.path(path)
 
         try:
+            # If a path does not exist, is_dir(), is_file() and is_symlink() will return False (and not raise an
+            # exception), so we need to explicitly check for this.
+            exists = self.target.fs.lexists(str(path))
```
Comment (Schamper) [inline acquire/collector.py:346]: I think there's one issue here currently, where if `path` is already a `TargetPath` but from a different filesystem. That might be an issue here: https://github.com/fox-it/acquire/blob/main/acquire/acquire.py#L256

I think you said that `path.get().lexists()` unfortunately doesn't work reliably, so then I guess it unfortunately has to become something like `path._fs.lexists(str(path))`?

Comment (pyrco) [reply acquire/collector.py:346]: Hmm, that won't work then. I'll temporarily use the `path._fs.lexists(str(path))` way then until the issues with `TargetPath.get()` and `FileEntry.lexists()` are resolved.

**Review by Schamper — DISMISSED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**